packages/labs/ssr/src/test/lib/render-lit_test.ts

```diff
@@ -423,6 +432,117 @@ for (const global of [emptyVmGlobal, shimmedVmGlobal]) {
+        tagName: 'dedupe-custom',
+      }),
+    });
+    assert.is(
+      removeDedupeScriptTagHelper(result),
+      `<!--lit-part sUH7hRvaZ8U=--><test-static-styles><template shadowroot="open" shadowrootmode="open">
+<style>
+  :host {
+    display: block;
+    background-color: blue;
+  }
+</style><dedupe-custom style-id="1" style="display:none;"></dedupe-custom><!--lit-part--><!--/lit-part--></template></test-static-styles
+  ><test-static-styles><template shadowroot="open" shadowrootmode="open"><dedupe-custom style-id="1" style="display:none;"></dedupe-custom><!--lit-part--><!--/lit-part--></template></test-static-styles><!--/lit-part-->`
+    );
+  });
+
+  test('element with static styles array using styles de-duplication utility', async () => {
+    const {render, sharingFromStaticStylesArray} = await setup();
+    const result = await render(sharingFromStaticStylesArray, {
+      dedupeStyles: new DeclarativeStyleDedupeUtility(),
+    });
+
+    assert.is(
+      removeDedupeScriptTagHelper(result),
+      `<!--lit-part vozTZ75mUhg=--><test-static-styles-array><template shadowroot="open" shadowrootmode="open">
+<style>
+  :host {
+    display: block;
+    background-color: blue;
+  }
+</style><lit-ssr-style-dedupe style-id="1" style="display:none;"></lit-ssr-style-dedupe><style>
+  :host {
+    margin: 8px;
+  }
+</style><lit-ssr-style-dedupe style-id="3" style="display:none;"></lit-ssr-style-dedupe><!--lit-part--><!--/lit-part--></template></test-static-styles-array><test-static-styles><template shadowroot="open" shadowrootmode="open"><lit-ssr-style-dedupe style-id="1" style="display:none;"></lit-ssr-style-dedupe><!--lit-part--><!--/lit-part--></template></test-static-styles><!--/lit-part-->`
+    );
+  });
+
+  test('doubled element with static styles array using deduplicated styles', async () => {
+    const {render, duplicatingElementWithStaticStylesArray} = await setup();
+    const result = await render(duplicatingElementWithStaticStylesArray, {
+      dedupeStyles: new DeclarativeStyleDedupeUtility(),
+    });
+
+    assert.is(
+      removeDedupeScriptTagHelper(result),
+      `<!--lit-part SoBZcCoc/o0=--><test-static-styles-array><template shadowroot="open" shadowrootmode="open">
+<style>
+  :host {
+    display: block;
+    background-color: blue;
+  }
+</style><lit-ssr-style-dedupe style-id="1" style="display:none;"></lit-ssr-style-dedupe><style>
+  :host {
+    margin: 8px;
+  }
+</style><lit-ssr-style-dedupe style-id="3" style="display:none;"></lit-ssr-style-dedupe><!--lit-part--><!--/lit-part--></template></test-static-styles-array><test-static-styles-array><template shadowroot="open" shadowrootmode="open"><lit-ssr-style-dedupe style-id="1" style="display:none;"></lit-ssr-style-dedupe><lit-ssr-style-dedupe style-id="3" style="display:none;"></lit-ssr-style-dedupe><!--lit-part--><!--/lit-part--></template></test-static-styles-array><!--/lit-part-->`
+    );
+  });
+
```

Comment:
Could we test that sharing the same instance of the dedupe utility allows deduping across multiple render calls?